onnxruntime/test/perftest/ort_test_session.cc

```diff
@@ -439,7 +439,7 @@ select from 'TF8', 'TF16', 'UINT8', 'FLOAT', 'ITENSOR'. \n)");
       }
     }
     // COREML_FLAG_CREATE_MLPROGRAM
-    Ort::ThrowOnError(OrtSessionOptionsAppendExecutionProvider_CoreML(session_options, coreml_flags));
+    session_options.AppendExecutionProvider("Coreml", {{"coreml_flags", std::to_string(coreml_flags)}});
```

Comment:
May be better to convert this is just pass through option key/value pairs instead of setting flags (assuming we add options that cover the current flags). That way it doesn't need an update in the future. 

We should have some shared parsing code though instead of so many different EPs parsing for a '|' to find the key and value. It's a little ridiculous that DNNL, CUDA, TensorRT, QNN, SNPE, DML, ACL and VitisAI are all repeating the same basic logic. Maybe ParseSessionConfigs in test/perftest/command_args_parser.cc can be used. Move it into a shared helper and call to create unordered_map<string, string> for the key/value pairs. The EP specific code can validate values using that map.